app/controllers/clearance/passwords_controller.rb

```diff
@@ -88,18 +87,19 @@ def flash_failure_after_update
       default: t('flashes.failure_after_update'))
   end
 
-  def forbid_missing_token
-    if params[:token].to_s.blank?
+  def check_for_forbidden_access
+    if missing_token? || user_does_not_exist?
```

Comment:
That's a great point! I suspect that the token check was initially put in place to kind of short-circuit the checks or something.